Marlin/Marlin_main.cpp

```diff
@@ -3761,10 +3768,23 @@ inline void gcode_G4() {
     sync_plan_position();
 
     // Move all carriages together linearly until an endstop is hit.
-    current_position[X_AXIS] = current_position[Y_AXIS] = current_position[Z_AXIS] = (Z_MAX_LENGTH + 10);
+    current_position[X_AXIS] = current_position[Y_AXIS] = current_position[Z_AXIS] = (DELTA_HEIGHT + home_offset[Z_AXIS] + 10);
```

Comment:
Guesstimating a height that minimally grinds the steppers in the event of endstops failure is a good thing. Just be sure this pertains to all possible configurations.